### PR TITLE
Automatically refresh metrics on web UI

### PR DESCRIPTION
**Pull request by luejerry**
- Auto update AQI metrics in UI at same interval as sensor pollrate
- Add basic JSON endpoint for AQI metrics at `/json`. Returns data in the following format:

```ts
{
  air_quality_index: {
    aqi_current: {
      value: float,
      color: string,
    },
    aqi_10min: { ... },
    aqi_1hour: { ... },
    aqi_24hour: { ... },
  },
  environment?: { // only if BME680 is available
    temperature: { value: float },
    pressure: { value: float },
    humidity: { value: float },
  },
}
```

### CODE REVIEW

File: src/Application.cpp [hunk 1]
```diff
@@ -292,6 +313,14 @@ String Application::processStatsPageHTML(const String& var)
 
   return String();
 }
+
+String Application::processScriptFile(const String &var) {
+  if (var == "SENSOR-UPDATE-SECONDS") {
+    return String(AIR_QUALITY_SENSOR_UPDATE_SECONDS);
```
Comment (michaelkamprath) [inline src/Application.cpp:349]: I would argue this is too fast in the default case. In the original configuration, that means the page would auto refresh every 2 seconds. While the ESP32 is a pretty decent MCU, I would not want to put that much load on it. I would suggest that the minimum value we use for an auto refresh be 1 minutes.

File: data/script.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  } else if (aqiValue <= 150) {
+    return 'aqi-orange';
+  } else if (aqiValue <= 200) {
+    return 'aqi-red';
+  } else if (aqiValue <= 300) {
+    return 'aqi-purple';
+  } else {
+    return 'aqi-maroon';
+  }
+}
```
Comment (michaelkamprath) [inline data/script.js]: This is what the original `COLOR-*` tokens were intended to cover, as they would get replaced by the `aui-*` CSS class names in the C++ `Application::processRootPageHTML()` method. By also implementing this mappingsg in JS, we now have two implementations of the same mapping, which could create its own class of maintenance problems over time. I would prefer we stick to the token placement approach, which keeps the function in the C++ as the logic is also needed to set the LED color (which must happen in the C++).

File: data/index.html [hunk 3]
```diff
@@ -11,23 +11,23 @@
     <div class="content">
       <div class="title">DIY Air Quality Monitor</div>
       <div id="current" class="aqi-display">
-        <div class="^COLOR-CURRENT^">
-          <div class="aqi-value">^AQI-CURRENT^</div>
+        <div class="^COLOR-CURRENT^" id="bg-current">
+          <div class="aqi-value" id="value-current">^AQI-CURRENT^</div>
         </div>
```
Comment (michaelkamprath) [inline data/index.html]: All these changes main to `index.html` need to also be made to `index_bme680.html` as that is the version of the index HTML that is used with the BME680 sensor is attached. Better yet, maybe consolidate everything to one index HTML file and have a method that strips out the BME680 divs if the sensor is not present.

File: src/Application.cpp [hunk 4]
```diff
@@ -181,6 +183,25 @@ void Application::handleStatsPageRequest(AsyncWebServerRequest *request)
   request->send(SPIFFS, stats_file, getContentType(stats_file), false, std::bind(&Application::processStatsPageHTML, this, std::placeholders::_1));
 }
 
+void Application::handleScriptRequest(AsyncWebServerRequest *request)
+{
+  String script_file = "/script.js";
+  request->send(SPIFFS, script_file, getContentType(script_file), false, std::bind(&Application::processScriptFile, this, std::placeholders::_1));
+}
```
Comment (michaelkamprath) [inline src/Application.cpp]: This function is missing the log line (`Serial.print(...)`), which strictly speaking isn't really needed but all the other handlers do log their web requests.

**Review by michaelkamprath — CHANGES_REQUESTED** [review]
Wow, thanks @luejerry ! This is cool. I do have some comments pertaining to how your changes fit into the overall architecture, but generally I like this.

File: src/Application.cpp (hunk 1 above)
Comment (luejerry) [reply src/Application.cpp:349]: Changed the refresh interval to 60s on the frontend. Currently hardcoded, but might add a widget in the future

File: data/script.js (hunk 2 above)
Comment (luejerry) [reply data/script.js]: Removed the frontend logic. The controller's `/json` endpoint now returns the appropriate CSS class name with each metric value.

File: data/index.html (hunk 3 above)
Comment (luejerry) [reply data/index.html]: I did as you suggested and consolidated them both into a single file, which should only show the environment data if the sensor is available.

**Comment by luejerry** [comment]
As part of consolidating the BME/non-BME pages, I went ahead and enabled auto refresh for BME metrics as well. Unfortunately I don't actually have the sensor so I'm not able to test whether it works, but it should auto update the environment values in the UI at the same time as the AQI metrics.

**Review by michaelkamprath — APPROVED** [review]

**Comment by michaelkamprath** [comment]
Thanks @luejerry !

**Pull request merged**